### PR TITLE
Fix build issue

### PR DESCRIPTION
**Pull request by aididhaiqal**

### CODE REVIEW

**Comment by aididhaiqal** [comment]
Promote / Demote is still in beta...Polish coming soon

File: CoCSharp/Network/Messages/ChatMessageServerMessage.cs [hunk 1]
```diff
@@ -95,16 +95,13 @@ public override void WriteMessage(MessageWriter writer)
             writer.Write(League);
             writer.Write(UserID);
             writer.Write(CurrentUserID);
-
+            writer.Write(Clan == null ? false : true);
```
Comment (FICTURE7) [inline CoCSharp/Network/Messages/ChatMessageServerMessage.cs]: I don't think that checking if the Clan is null twice is a good idea.

Comment (aididhaiqal) [inline CoCSharp/Network/Messages/ChatMessageServerMessage.cs]: Yep?Changed in my local...Will push within 5 minute as currently updating network driver on oc

Comment (aididhaiqal) [inline CoCSharp/Network/Messages/ChatMessageServerMessage.cs]: I mean yep without ?

**Comment by NSbuilder** [comment]
aidid? I thought you are working on GCS...

File: CoCSharp/Network/Messages/AllianceStream/ChatStreamEntry.cs [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+        public int League;
+        /// <summary>
+        /// User Level
+        /// </summary>
+        public ClanMemberRole Role;
+        /// <summary>
+        /// UserName
+        /// </summary>
+        public string Message;
+        private DateTime m_vMessageTime;
```
Comment (FICTURE7) [inline CoCSharp/Network/Messages/AllianceStream/ChatStreamEntry.cs]: Convention please mate... And why is it private?

File: CoCSharp/Network/Messages/AllianceStream/ChatStreamEntry.cs [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+            ThrowIfWriterNull(writer);
+            writer.Write(MessageID);
+            writer.Write(Unknown1);
+            writer.Write(UserID);
+            writer.Write(HomeID);
+            writer.Write(Name);
+            writer.Write(Level);
+            writer.Write(League);
+            writer.Write((int)Role);
+            writer.Write((int)DateTime.UtcNow.Subtract(new DateTime(1970, 1, 1)).TotalSeconds - (int)m_vMessageTime.Subtract(new DateTime(1970, 1, 1)).TotalSeconds);
```
Comment (FICTURE7) [inline CoCSharp/Network/Messages/AllianceStream/ChatStreamEntry.cs]: Use DateTimeConverter instead.

File: CoCSharp/Network/Messages/AllianceStream/ChatStreamEntry.cs [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+    /// </summary>
+    public class ChatStreamEntry : AllianceStreamEntry
+    {
+        /// <summary>
+        /// Initializes a new instance of the <see cref="ChatStreamEntry"/> class.
+        /// </summary>
+        public ChatStreamEntry()
+        {
+            //space
+            Unknown1 = 3;
```
Comment (FICTURE7) [inline CoCSharp/Network/Messages/AllianceStream/ChatStreamEntry.cs]: We're not suppose to initialize anything in the default constructor of a Message/StreamEntry.

Comment (aididhaiqal) [inline CoCSharp/Network/Messages/AllianceStream/ChatStreamEntry.cs]: Ok will fix

File: CoCSharp/Network/Messages/AllianceStream/ChatStreamEntry.cs [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+        // <summary>
+        /// Reads the <see cref="ChatStreamEntry"/> from the specified <see cref="MessageReader"/>.
+        /// </summary>
+        /// <param name="reader">
+        /// <see cref="MessageReader"/> that will be used to read the <see cref="TroopRequestStreamEntry"/>.
+        /// </param>
+        /// <exception cref="ArgumentNullException"><paramref name="reader"/> is null.</exception>
+        public override void ReadStreamEntry(MessageReader reader)
+        {
+            ThrowIfReaderNull(reader);
```
Comment (FICTURE7) [inline CoCSharp/Network/Messages/AllianceStream/ChatStreamEntry.cs:79]: Implementing the reading part would be great.

**Pull request merged**